src/hotspot/os/linux/waitBarrier_linux.cpp

```diff
@@ -38,6 +38,13 @@
 
 #define guarantee_with_errno(cond, msg) check_with_errno(guarantee, cond, msg)
 
+// 32-bit RISC-V has no SYS_futex syscall.
+#ifndef SYS_futex
+  #if defined(RISCV32) && defined(SYS_futex_time64)
+    #define SYS_futex SYS_futex_time64
```

Comment:
Should it be better to check SYS_futex and SYS_futex_time64 when RISCV32 defined?

Reply:
Fixed.